webui/react/src/pages/ExperimentDetails/ExperimentCodeViewer.tsx

```diff
@@ -0,0 +1,104 @@
+import yaml from 'js-yaml';
+import React, { useMemo, useState } from 'react';
+
+import Icon from 'components/kit/Icon';
+import { paths } from 'routes/utils';
+import { getExperimentFileFromTree, getExperimentFileTree } from 'services/api';
+import { V1FileNode } from 'services/api-ts-sdk';
+import Spinner from 'shared/components/Spinner/Spinner';
+import { RawJson } from 'shared/types';
+import { ExperimentBase, TreeNode } from 'types';
+import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
+
+const CodeEditor = React.lazy(() => import('components/kit/CodeEditor'));
+
+const configIcon = <Icon name="settings" title="settings" />;
+
+export interface Props {
+  experiment: ExperimentBase;
+  onSelectFile?: (arg0: string) => void;
+  selectedFilePath?: string;
+}
+
+const ExperimentCodeViewer: React.FC<Props> = ({
+  experiment,
+  onSelectFile,
+  selectedFilePath,
+}: Props) => {
+  const [expFiles, setExpFiles] = useState<Loadable<TreeNode[]>>(NotLoaded);
+
+  const submittedConfig = useMemo(() => {
+    if (!experiment.originalConfig) return;
+
+    const { hyperparameters, ...restConfig } = yaml.load(experiment.originalConfig) as RawJson;
+
+    // don't ask me why this works.. it gets rid of the JSON though
+    return yaml.dump({ ...restConfig, hyperparameters });
+  }, [experiment.originalConfig]);
+
+  const runtimeConfig = useMemo(() => {
+    if (!experiment.configRaw) return;
+
+    const {
+      environment: { registry_auth, ...restEnvironment },
+      workspace,
+      project,
+      ...restConfig
+    } = experiment.configRaw;
+    return yaml.dump({ environment: restEnvironment, ...restConfig });
+  }, [experiment.configRaw]);
+
+  useMemo(async () => {
+    const convertV1FileNodeToTreeNode = (node: V1FileNode): TreeNode => ({
+      children: node.files?.map((n) => convertV1FileNodeToTreeNode(n)) ?? [],
+      content: NotLoaded,
+      download: paths.experimentFileFromTree(experiment.id, String(node.path)),
+      get: (path: string) => getExperimentFileFromTree({ experimentId: experiment.id, path }),
+      isLeaf: !node.isDir,
+      key: node.path ?? '',
+      title: node.name,
+    });
+
+    const fileTree = await getExperimentFileTree({ experimentId: experiment.id });
+    setExpFiles(Loaded(fileTree.map<TreeNode>(convertV1FileNodeToTreeNode)));
+  }, [experiment.id]);
```

Comment:
sorry, just realized this should be an effect.

Reply:
would the same be true for `experiment.submittedConfig` and `experiment.runtimeConfig` -- I don't expect any of these properties to be modified or updated while we're on these page